controllers/storagecluster/placement.go

```diff
@@ -27,6 +27,15 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 		placement.PodAntiAffinity = &corev1.PodAntiAffinity{}
 	}
 
+	// if number of mds or rgw pods is greater than number of failure domains, then disable requiredDuringScheduling pod anti-affinity
+	failureDomains := len(sc.Status.FailureDomainValues)
+	if (component == "mds" && 2*getActiveMetadataServers(sc) > failureDomains) || (component == "rgw" && getCephObjectStoreGatewayInstances(sc) > failureDomains) {
+		if placement.PodAntiAffinity != nil {
+			placement.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = nil
```

Comment:
I wonder if it would be sufficient to just always set preferred anti-affinity. Then we don't need complex calculations like this to figure out if we should have required affinity. I don't see a real need to use required anti-affinity.

Reply:
Actually, I saw the preferred anti-affinity is not really that reliable. While trying to test this on my setup, I tried to increase the active MDS count to 3, which would bring 6 pods total. I have 3 nodes, and the pods came up like 3+2+1. So there is a risk of skewing of pods towards a particular node, which someone might not like. So I think we should skip the required anti-affinity only when it's required.

Reply:
If there are more daemons than the number of nodes, the preferred anti-affinity does nothing to spread the daemons evenly, that is correct. We would need to use topology spread constraints if we cared about even spread. But at least with preferred anti-affinity, it allows additional daemons to be scheduled. And when there are more daemons than nodes, there is at least one daemon per node. If there is no anti-affinity, then the daemons could all end up on the same node, which is worse than at least one per node.

Reply:
So we should be good with removing the required anti affinity all together in case of mds & rgw?